common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -493,6 +496,49 @@ protected Path getRecoveryPath(String fileName) {
     return _recoveryPath;
   }
 
+  private void migrateRecoveryDb() throws IOException {
+    Preconditions.checkNotNull(_recoveryPath,
+            "recovery path should not be null if NM recovery is enabled");
+
+    DBBackend registeredExecutorsDbBackend = findExistedDbBackend(RECOVERY_FILE_NAME);
```

Comment:
line 503 ~ line 511 can be extract  into a general method and put into the helper class